closed/make/Javadoc.gmk

```diff
@@ -483,3 +479,268 @@ $(OPENJ9_JAVADOC_MARKER) : $(COREAPI_DOCSDIR)/index.html $(OPENJ9_MGMT_INDEXALL_
 
 # ALL_OTHER_TARGETS are declared as PHONY in make/Javadoc.gmk
 ALL_OTHER_TARGETS += $(OPENJ9_JAVADOC_MARKER) openj9-sharedclassesdocs openj9-gpudocs openj9-cudadocs openj9-dataaccessdocs openj9-jvmdocs
+
+#############################################################################
+# Generate javadoc from only the openj9 source files with reference links to
+# the online Oracle javadoc for the classes derived from openjdk source
+# files.
+#############################################################################
+
+#############################################################################
+# Add an output directory to put the openj9 only javadoc.
+OPENJ9_ONLY_DOCSDIR=$(OUTPUT_ROOT)/openj9-docs
+
+OPENJ9_ONLY_PLATFORM_DOCSDIR = $(OPENJ9_ONLY_DOCSDIR)/platform
+
+#############################################################################
+# Set the link prefixes for the openjdk source file classes to reference the
+# online Oracle documentation.
+OPENJ9_ONLY_JAVASE_BASE_URL := https://docs.oracle.com/javase/8/docs/api/
+OPENJ9_ONLY_MGMT_BASE_URL := https://docs.oracle.com/javase/8/docs/jre/api/management/extension/
+
+#############################################################################
+# Exclude the openjdk profiles data.
+OPENJ9_ONLY_COMMON_JAVADOCFLAGS = $(filter-out -Xprofilespath $(JDK_TOPDIR)/make/profile-rtjar-includes.txt,$(COMMON_JAVADOCFLAGS))
+
+#############################################################################
+#
+# openj9-only-sharedclassesdocs
+#
+
+OPENJ9_ONLY_SHAREDCLASSES_DOCDIR := $(OPENJ9_ONLY_PLATFORM_DOCSDIR)/sharedclasses
+
+# The index.html and options files
+# The package file generated for the full jdk javadoc build is reused.
+OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE := $(OPENJ9_ONLY_SHAREDCLASSES_DOCDIR)/index.html
+OPENJ9_ONLY_SHAREDCLASSES_OPTIONS_FILE := $(DOCSTMPDIR)/openj9_only.sharedclasses.options
+
+openj9-only-sharedclassesdocs: $(OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE)
```

Comment:
Are you saying you'd like to see
```
openj9-only-sharedclassesdocs : $(OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE)
```
rather than
```
openj9-only-sharedclassesdocs: $(OPENJ9_ONLY_SHAREDCLASSES_INDEX_FILE)
```
?

Reply:
Yes. Perhaps I'm being pedantic, but putting spaces on both sides of colons avoids the question of what this means:
```
C:/foo:bar
```
Is it
```
C : /foo:bar
```
or
```
C:/foo : bar
```
All those names are possible (even if unlikely, particularly on Windows). The separation makes it clear (most importantly to humans).

Reply:
OK, but the other targets in the file do not have a space either side of the colon.  I think they should be consistent within a makefile.  (I'm happy to make them all have a space either side of the colon, if that's the standard we're adopting).

Reply:
Main.gmk had those spaces; I think we should aim to apply this rule to all makefiles within the 'closed' folder.

Reply:
I've update to include a space on either side of a target's colon.